flytekit/core/pod_template.py

```diff
@@ -0,0 +1,22 @@
+from dataclasses import dataclass
+from typing import Dict, Optional
+
+from kubernetes.client.models import V1PodSpec
+
+from flytekit.exceptions import user as _user_exceptions
+
+PRIMARY_CONTAINER_DEFAULT_NAME = "primary"
+
+
+@dataclass
+class PodTemplate(object):
+    pod_spec: V1PodSpec
+    primary_container_name: str = PRIMARY_CONTAINER_DEFAULT_NAME
```

Comment:
This is going to be a point of contention. We should probably rename this to something like `task_container_name` and have it default to something other than `primary`.

This is because the `primary_container_name` in the `Pod` `task_config` is very specific, and not only for applying configuration flytekit-side. On the backend, if the `primary_container_name` is specified through `TaskTemplate.Config` we declare the task completed when the primary container completes. This allows The `Pod` `task_config` to inject sidecars that run indefinitely. However, this is the opposite of what is necessary for `ContainerTasks` to complete. Where we inject a container(s) (copilot) to download inputs and then output the task outputs. For a task to complete, all of the containers need to completed (rather than just the primary container in the case of `Pod` `task_config`). So if we want to support this `PodTemplate` from `ContainerTasks` (which we certainly do), we need to say that all containers need to complete for the task to complete. I think using the same naming (ie. `primary_container_name`) as the `Pod` `task_config` would make this confusing as the functionality is different.

Reply:
TL;DR IMO we should require all containers specified in the `PodTemplate` to complete for the task to complete. This is necessary if we want to support this functionality from the `ContainerTask`. The only thing we lose is the ability to apply this with the `Pod` `task_config`. But since the `Pod` task already defines a `Pod` it already has the same support that this provides, so there is no need.

Reply:
I don't really understand this. Seems to lack some context. Will contact you offline.